src/python/gem5/components/processors/abstract_core.py

```diff
@@ -120,3 +120,35 @@ def get_mmu(self) -> BaseMMU:
         This is used in the board to setup system-specific MMU settings.
         """
         raise NotImplementedError
+
+    @abstractmethod
+    def set_simpoint(
+        self, inst_starts: List[int] = [], init: bool = True
+    ) -> None:
+        """Schedule simpoint exit events for the core.
+
+        This is used to raise SIMPOINT_BEGIN exit events in the gem5 standard
+        library.
+
+        :param inst_starts: a list of SimPoints starting instructions
+        :param init: if it is True, the starting instructions will be scheduled
+        at the init stage of the core, else, the starting insructions will be
+        scheduled during the simulation
+        """
+        raise NotImplementedError("This core type does not support simpoints")
+
+    @abstractmethod
+    def set_inst_stop_any_thread(
+        self, inst: int = 0, init: bool = True
```

Comment:
Why do these parameters have default values? Are neither of them required? `inst` seems required at a minimum as it doesn't make sense to stop at instruction 0, right?